stac/src/main/scala/package.scala

```diff
@@ -1,105 +1,157 @@
 package geotrellis.server
 
+import java.time.Instant
+
 import cats.implicits._
+import com.github.tbouron.SpdxLicense
+import eu.timepit.refined.api.{Refined, RefinedTypeOps, Validate}
+import eu.timepit.refined.collection.Exists
 import geotrellis.vector.{io => _, _}
 import io.circe._
 import io.circe.parser.{decode, parse}
 import io.circe.shapes.CoproductInstances
 import shapeless._
 
-import java.text.ParseException
-import java.time.Instant
-
 package object stac {
+
+  case class LicenseLink()
+
+  object LicenseLink {
+    implicit def validateLicenseLink: Validate.Plain[StacLink, LicenseLink] =
+      Validate.fromPredicate(
+        l =>
+          l.rel match {
+            case License => true
+            case _       => false
+          },
+        t => s"Not a License Link: ${t.rel}",
+        LicenseLink()
+      )
+  }
+
+  type StacLinksWithLicense = List[StacLink] Refined Exists[LicenseLink]
+  object StacLinksWithLicense
+      extends RefinedTypeOps[StacLinksWithLicense, List[StacLink]]
```

Comment:
I didn't know this until last night, but using type aliases in this way lets us do something like:

```scala
val result: Either[Left, Right(StacLinksWithLicense)] = StacLinksWithLicense.from(x: List[StacLink])
```

which I think is a totally acceptable requirement for consumers of this library to deal with

Reply:
Sounds reasonable to me. It also looks like `RefinedTypeOps` provides an escape hatch for runtime checks, allowing `StacLinksWithLicense.apply(x: List[StacLink])`.

Reply:
Nevermind. Looks like the apply is available via macros and you've got to call `unsafeFrom` to avoid compile-time checks